src/expression.d

```diff
@@ -637,31 +637,56 @@ extern (C++) Expression resolvePropertiesOnly(Scope* sc, Expression e1)
  */
 extern (C++) Expression searchUFCS(Scope* sc, UnaExp ue, Identifier ident)
 {
+    //printf("searchUFCS(ident = %s)\n", ident.toChars());
     Loc loc = ue.loc;
-    Dsymbol s = null;
-    for (Scope* scx = sc; scx; scx = scx.enclosing)
+
+    // TODO: merge with Scope.search.searchScopes()
+    Dsymbol searchScopes(int flags)
     {
-        if (!scx.scopesym)
-            continue;
-        s = scx.scopesym.search(loc, ident);
-        if (s)
+        Dsymbol s = null;
+        for (Scope* scx = sc; scx; scx = scx.enclosing)
         {
-            // overload set contains only module scope symbols.
-            if (s.isOverloadSet())
-                break;
-            // selective/renamed imports also be picked up
-            if (AliasDeclaration ad = s.isAliasDeclaration())
+            if (!scx.scopesym)
+                continue;
+            if (scx.scopesym.isModule())
+                flags |= SearchUnqualifiedModule;    // tell Module.search() that SearchLocalsOnly is to be obeyed
+            s = scx.scopesym.search(loc, ident, flags);
+            if (s)
             {
-                if (ad._import)
+                // overload set contains only module scope symbols.
+                if (s.isOverloadSet())
+                    break;
+                // selective/renamed imports also be picked up
+                if (AliasDeclaration ad = s.isAliasDeclaration())
+                {
+                    if (ad._import)
+                        break;
+                }
+                // See only module scope symbols for UFCS target.
+                Dsymbol p = s.toParent2();
+                if (p && p.isModule())
                     break;
             }
-            // See only module scope symbols for UFCS target.
-            Dsymbol p = s.toParent2();
-            if (p && p.isModule())
+            s = null;
+
+            // Stop when we hit a module, but keep going if that is not just under the global scope
+            if (scx.scopesym.isModule() && !(scx.enclosing && !scx.enclosing.enclosing))
                 break;
         }
-        s = null;
+        return s;
+    }
+
+    Dsymbol s;
+
+    if (global.params.bug10378)
+        s = searchScopes(0);
+    else
```

Comment:
It isn't quite that simple, but you're telling people that to use the old way they have to throw a switch, same as with this PR.